RecoHI/HiJetAlgos/plugins/HiFJRhoFlowModulationProducer.cc

```diff
@@ -90,128 +172,202 @@ void HiFJRhoFlowModulationProducer::produce(edm::Event& iEvent, const edm::Event
+  const int nParamVals =
+      nFlow * 2 +
+      4;  // v_n and Psi_n for each fitted flow component, plus overall normalization factor, chi^2 and ndf for flow fit, and the first fitted flow component
   auto rhoFlowFitParamsOut = std::make_unique<std::vector<double>>(nParamVals, 1e-6);
 
-  rhoFlowFitParamsOut->at(0) = 0;
-  rhoFlowFitParamsOut->at(1) = 0;
-  rhoFlowFitParamsOut->at(2) = 0;
-  rhoFlowFitParamsOut->at(3) = 0;
-  rhoFlowFitParamsOut->at(4) = 0;
-
-  double eventPlane2Cos = 0;
-  double eventPlane2Sin = 0;
+  // Set the parameters related to flow fit to zero
+  for (int iParameter = 0; iParameter < nFlow * 2 + 1; iParameter++) {
+    rhoFlowFitParamsOut->at(iParameter) = 0;
+  }
+  // Set the first fitted flow component to the last index of the output vector
+  rhoFlowFitParamsOut->at(nFlow * 2 + 3) = firstFittedVn_;
 
-  double eventPlane3Cos = 0;
-  double eventPlane3Sin = 0;
+  // Initialize arrays for manual event plane calculation
+  double eventPlaneCos[nFlow];
+  double eventPlaneSin[nFlow];
+  for (int iFlow = 0; iFlow < nFlow; iFlow++) {
+    eventPlaneCos[iFlow] = 0;
+    eventPlaneSin[iFlow] = 0;
+  }
 
+  // Define helper variables for looping over particle flow candidates
   std::vector<bool> pfcuts(pfCands.size(), false);
   int iCand = -1;
+  double thisPfCandidateWeight = 0;
+  double deltaPhiJetPf = 0;
+  std::vector<double> pfCandidateWeight(pfCands.size(), 1);
+
+  // Loop over the particle flow candidates
   for (auto const& pfCandidate : pfCands) {
-    iCand++;
-    if (pfCandidate.particleId() != 1 || std::abs(pfCandidate.eta()) > 1.0 || pfCandidate.pt() < .3 ||
-        pfCandidate.pt() > 3.) {
+    iCand++;  // Update the particle flow candidate index
+
+    // Find the PF candidate ID from the packed candidates collection
+    auto particleFlowCandidateID = converter_.translatePdgIdToType(pfCandidate.pdgId());
+
+    // This cut selects particle flow candidates that are charged hadrons. The ID numbers are:
+    // 0 = undefined
+    // 1 = charged hadron
+    // 2 = electron
+    // 3 = muon
+    // 4 = photon
+    // 5 = neutral hadron
+    // 6 = HF tower identified as a hadron
+    // 7 = HF tower identified as an EM particle
+    if (particleFlowCandidateID != 1)
       continue;
-    }
 
+    // Kinematic cuts for particle flow candidate pT and eta
+    if (std::abs(pfCandidate.eta()) > pfCandidateEtaCut_)
+      continue;
+    if (pfCandidate.pt() < pfCandidateMinPtCut_)
+      continue;
+    if (pfCandidate.pt() > pfCandidateMaxPtCut_)
+      continue;
+
+    nFill++;  // Use same nFill criterion with and without jetty region subtraction
+
+    // If the jetty regions are excluded from the fit, find which particle flow candidates are close to jets
+    thisPfCandidateWeight = 1;
     if (doJettyExclusion_) {
       bool isGood = true;
       for (auto const& jet : *jets) {
-        if (deltaR2(jet, pfCandidate) < .16) {
+        if (deltaR2(jet, pfCandidate) < 0.16) {
```

Comment:
Does this hard-coded 0.16 assume that the jet radius is 0.4?  
Should it depend on the R value used to cluster the jet?
Should it be configurable?